lib/spack/spack/cmd/create.py

```diff
@@ -328,65 +353,169 @@ def __call__(self, stage, url):
 
         # Determine the build system based on the files contained
         # in the archive.
-        build_system = 'unknown'
+        build_system = 'default'
```

Comment:
So the name is only presented to the user in 2 scenarios. One, as a help option for `spack create --template`, and two, when it prints out "This package looks like it uses the {0} build system". I think `--template generic` makes more sense, but I also think `the generic build system` doesn't make much sense. I could reword it of course.